src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -355,6 +355,16 @@ private bool PrunePackages(PublishRoot root)
                             keep.Add(CombinePath(packagesDir, path));
                         }
 
+                        // Handling special case like LibGit2Sharp.0.21.0.176, which contains
+                        // lib/net40/NativeBinaries/x86/git2-e0902fb.dll
+                        var runtimeAssembly = library.RuntimeAssemblies.FirstOrDefault();
+                        if (runtimeAssembly != null)
+                        {
+                            var assemblyFolder = Path.GetDirectoryName(CombinePath(packagesDir, runtimeAssembly));
+                            keep.AddRange(Directory.EnumerateFiles(assemblyFolder, "*.*", SearchOption.AllDirectories)
+                                .Where(f => f.EndsWith(".dll", StringComparison.OrdinalIgnoreCase)));
+                        }
+
```

Comment:
@davidfowl , sorry I didn't notice `LibGit2Sharp.0.21.0.176` is a different special case. Just added this part, so that
`LibGit2Sharp\0.21.0.176\lib\net40\NativeBinaries\amd64\git2-e0902fb.dll`
`LibGit2Sharp\0.21.0.176\lib\net40\NativeBinaries\x86\git2-e0902fb.dll`
will get preserved during publish.

Let me know whether you like the logic here.